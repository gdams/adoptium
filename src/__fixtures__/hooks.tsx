import {
  Binary,
  ContributorApiResponse,
  Contributor,
  LatestTemurin,
  MarketplaceRelease,
  News,
  ReleaseAsset,
  TemurinReleases,
} from '../hooks';

export const createRandomContributorApiData = (): ContributorApiResponse => ({
  login: 'login_mock',
  id: 0,
  node_id: 'node_id_mock',
  avatar_url: 'avatar_url_mock',
  gravatar_id: 'gravatar_id_mock',
  url: 'url_mock',
  html_url: 'html_url_mock',
  followers_url: 'followers_url_mock',
  following_url: 'following_url_mock',
  gists_url: 'gists_url_mock',
  starred_url: 'starred_url_mock',
  subscriptions_url: 'subscriptions_url_mock',
  organizations_url: 'organizations_url_mock',
  repos_url: 'repos_url_mock',
  events_url: 'events_url_mock',
  received_events_url: 'received_events_url_mock',
  type: 'type_mock',
  site_admin: false,
  contributions: 3,
});

export const createRandomTemurinRelease = (installer, id): ReleaseAsset => ({
  os: 'os_mock',
  architecture: 'arch_mock',
  release_link: new URL('https://release_link_mock'),
  platform_name: `platform_name_mock${id}`,
  release_name: 'release_name_mock',
  release_date: new Date(Date.UTC(2020, 0, 1)),
  binaries: [
    {
      type: 'type_mock',
      link: new URL('https://link_mock'),
      checksum: 'checksum_mock',
      size: 0,
      extension: 'extension_mock',
      installer_link: installer ? new URL('https://installer_link_mock') : undefined,
      installer_checksum: installer ? 'installer_checksum_mock' : undefined,
      installer_size: installer ? 0 : undefined,
      installer_extension: installer ? 'installer_extension_mock' : undefined,
    },
  ],
})

export const createRandomMarketplaceRelease = (installer): MarketplaceRelease => ({
  release_name: 'release_name_mock',
  vendor: 'vendor_mock',
  binary: {
    os: 'os_mock',
    architecture: 'arch_mock',
    distribution: 'distribution_mock',
    image_type: 'type_mock',
    jvm_impl: 'jvm_impl_mock',
    timestamp: new Date(Date.UTC(2020, 0, 1)),
    scm_ref: 'scm_ref_mock',
    openjdk_scm_ref: 'openjdk_scm_ref_mock',
    aqavit_results_link: new URL('https://aqavit_results_link_mock'),
    tck_affidavit_link: new URL('https://tck_affidavit_link_mock'),
    package: {
      name: 'name_mock.tar.gz',
      link: new URL('https://link_mock'),
      sha265sum: 'sha265sum_mock',
      sha256sum_link: new URL('https://sha256sum_link_mock'),
      signature_link: new URL('https://signature_link_mock'),
    },
    installer: installer
      ? [
          {
            name: 'installer_name_mock.msi',
            link: new URL('https://installer_link_mock'),
            sha265sum: 'installer_sha265sum_mock',
            sha256sum_link: new URL('https://installer_sha256sum_link_mock'),
            signature_link: new URL('https://installer_signature_link_mock'),
          },
        ]
      : undefined,
  },
  version: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  }
})

export const mockLatestTemurin = (installer): LatestTemurin => ({
  download_count: 0,
  id: 'id_mock',
  release_link: new URL('https://release_link_mock'),
  release_name: 'release_name_mock',
  release_type: 'release_type_mock',
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  updated_at: new Date(Date.UTC(2020, 0, 1)),
  vendor: 'vendor_mock',
  version_data: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  },
  binaries: [
    {
      os: 'os_mock',
      architecture: 'arch_mock',
      image_type: 'type_mock',
      heap_size: 'heap_size_mock',
      download_count: 0,
      jvm_impl: 'jvm_impl_mock',
      package: {
        name: 'name_mock.tar.gz',
        link: new URL('https://link_mock'),
        checksum: 'sha265sum_mock',
        checksum_link: new URL('https://sha256sum_link_mock'),
        metadata_link: new URL('https://metadata_link_mock'),
        signature_link: new URL('https://signature_link_mock'),
        size: 0,
        download_count: 0,
      },
      installer: installer
      ?
        {
          name: 'installer_name_mock.msi',
          link: new URL('https://installer_link_mock'),
          checksum: 'installer_sha265sum_mock',
          checksum_link: new URL('https://installer_sha256sum_link_mock'),
          metadata_link: new URL('https://installer_metadata_link_mock'),
          signature_link: new URL('https://installer_signature_link_mock'),
          size: 0,
          download_count: 0,
        }
      : undefined,
    }
  ]
})

export const createRandomTemurinReleases = (installer, id): TemurinReleases => ({
  release_name: `release_name_mock_${id}`,
  release_link: new URL('https://release_link_mock'),
  source_url: new URL('https://source_url_mock'),
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  platforms: {
    'platform_mock': {
      assets: [
        {
          os: 'os_mock',
          architecture: 'architecture_mock',
          type: 'type_mock',
          link: new URL('https://link_mock'),
          checksum: `checksum_mock${id}`,
          size: 0,
          extension: 'extension_mock',
          installer_link: installer ? new URL('https://installer_link_mock') : undefined,
          installer_checksum: installer ? 'installer_checksum_mock' : undefined,
          installer_size: installer ? 0 : undefined,
          installer_extension: installer ? 'installer_extension_mock' : undefined,
        },
      ],
    },
  },
})

export const createRandomContributorViewData = (): Contributor => ({
  avatarUri: 'avatarUri_mock',
  profileUri: 'profileUri_mock',
  login: 'login_mock',
  contributionsCount: 0,
  commitsListUri: 'commitsListUri_mock',
  repo: 'reponame_mock',
});

export const createRandomLatestForOSData = (): Binary => ({
  release_name: 'release_name_mock',
  link: new URL('https://link_mock'),
});

export const createRandomNewsAndEventsData = (): News => ({
  news: [
    {
      id: 'id_mock',
      title: 'title_mock',
      body: 'body_mock',
      date: new Date(Date.UTC(2020, 0, 1)),
      link: 'link_mock',
    },
  ],
  events: [
    {
      id: 'id_mock',
      title: 'title_mock',
      infoLink: 'infoLink_mock',
      date: new Date(Date.UTC(2020, 0, 1)),
    },
  ]
});

export const createRandomContributorViewData1Contribution = (): Contributor => ({
  avatarUri: 'avatarUri_mock',
  profileUri: 'profileUri_mock',
  login: 'login_mock',
  contributionsCount: 1,
  commitsListUri: 'commitsListUri_mock',
  repo: 'reponame_mock',
});