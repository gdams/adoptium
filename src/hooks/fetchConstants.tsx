import { useEffect, useState } from 'react';
import axios from 'axios';

const baseUrl = 'https://staging-api.adoptium.net/v3';

export function fetchOses(isVisible: boolean): String[] {

    const [oses, setOses] = useState<String[]>([]);

    useEffect(() => {
        if (isVisible) {
        (async () => {
            const url = `${baseUrl}/info/available_operating-systems`;

            axios.get(url)
                .then(function (response) {
                    setOses(response.data);
                })
                .catch(function (error) {
                    setOses([]);
                });
        })();
        }
    }, [isVisible]);

    return oses;
}

export function fetchArches(isVisible: boolean): String[] {

    const [arches, setArches] = useState<String[]>([]);

    useEffect(() => {
        if (isVisible) {
        (async () => {
            const url = `${baseUrl}/info/available_architectures`;

            axios.get(url)
                .then(function (response) {
                    const results:String[] = response.data;
                    const idx = results.findIndex(it => it === 'x32');
                    if(idx >= 0) results.splice(idx, 1, 'x86');
                    setArches(results);
                })
                .catch(function (error) {
                    setArches([]);
                });
        })();
        }
    }, [isVisible]);

    return arches;
}
