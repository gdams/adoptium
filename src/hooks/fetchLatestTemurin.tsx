import { useEffect, useState } from 'react';

const baseUrl = 'https://api.adoptium.net/v3';

export function fetchLatestForOS(
    isVisible: boolean,
    version: number,
    os: string,
    arch: string
): Binary | null {
    if (!os) {
        return null
    }
    const [binary, setBinary] = useState<Binary | null>(null);
    useEffect(() => {
        if (isVisible) {
        (async () => {
            setBinary(await fetchLatestForOSRequest(version, os, arch));
        })();
        }
    }, [isVisible]);

    return binary;
}

async function fetchLatestForOSRequest(version: number, os: string, arch: string) {
    const url = `${baseUrl}/assets/feature_releases/${version}/ga?os=${os}&architecture=${arch}&image_type=jdk&jvm_impl=hotspot&page_size=1&vendor=eclipse`;
    const response = await fetch(url);
    const json = (await response.json())[0];
    let binary_link = json.binaries[0].package.link
    if (json.binaries[0].installer) {
        binary_link = json.binaries[0].installer.link
    }
    return {
        release_name: json.release_name,
        link: binary_link
    };
}

export interface Binary {
  release_name: string;
  link: URL;
}