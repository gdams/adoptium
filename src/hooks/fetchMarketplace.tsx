const baseUrl = 'https://marketplace-api.adoptium.net';
import { VersionMetaData } from ".";
    
export async function getAllPkgsForVersion(
    version: number,
    os: string,
    architecture: string,
    package_type: string,
    vendors: string[]= [],
): Promise<MarketplaceRelease[] | null> {
    let params = '?'
    params += 'feature_version=' + version;

    if (os !== 'any' ) {
        if (os === 'alpine-linux') {
            params += '&os=alpine_linux'
        } else {
            params += ('&os=' + os)
        }
    }

    if (architecture !== 'any' ) {
        params += ('&architecture=' + architecture)
    }

    if (package_type === 'any') {
        params += '&image_type=jdk&image_type=jre'
    } else {
        params += ('&image_type=' + package_type)
    }

    for(const vendor of vendors) {
        params += ('&vendor=' + vendor)
    }

    const url = new URL(baseUrl + '/v1/assets/latestForVendors' + params);
    const data = await getPkgs(url);
    return data
}

async function getPkgs(url: URL) {
    const response = await fetch(url)
    return response.json();
}

export function getImageForDistribution(distribution: string) {
    switch(distribution) {
        case 'temurin': return '/images/adoptium-logo.png';
        case 'redhat': return '/images/redhat.svg';
        case 'bisheng': return '/images/huawei.svg';
        case 'zulu': return '/images/azul-logo.png';
        case 'semeru': return '/images/ibm-logo.png';
        default: return `/images/${distribution}-logo.png`;
    }
}

export interface MarketplaceRelease {
    release_name: string;
    vendor: string;
    binary: {
        os: string;
        architecture: string;
        image_type: string;
        jvm_impl: string;
        package: {
            name: string;
            link: URL;
            sha265sum: string;
            sha256sum_link: URL;
            signature_link: URL;
        }
        installer?: [{
            name: string;
            link: URL;
            sha265sum: string;
            sha256sum_link: URL;
            signature_link: URL;
        }]
        timestamp: Date;
        scm_ref: string;
        openjdk_scm_ref: string;
        distribution: string;
        aqavit_results_link: URL;
        tck_affidavit_link: URL;
    }
    version: VersionMetaData;
}
