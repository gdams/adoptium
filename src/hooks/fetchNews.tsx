import { useEffect, useState } from 'react';

const baseUrl = 'https://newsroom.eclipse.org/api';

export function fetchNewsItems(
    isVisible: boolean,
): News | null {
    const [news, setNews] = useState<NewsItem[] | null>([]);
    const [events, setEvents] = useState<EventItem[] | null>([]);
    useEffect(() => {
        if (isVisible) {
        (async () => {
            setNews(await fetchLatestNews());
            setEvents(await fetchLatestEvents());
        })();
        }
    }, [isVisible]);

    const newsAndEvents: News = {
        news: news,
        events: events
    }

    return newsAndEvents;
}

async function fetchLatestNews() {
    const url = `${baseUrl}/news?parameters[publish_to]=adoptium`;
    const response = await fetch(url);
    const json = (await response.json());
    return json.news
}

async function fetchLatestEvents() {
    const url = `${baseUrl}/events?parameters[publish_to]=adoptium`;
    const response = await fetch(url);
    const json = (await response.json());
    return json.events
}

export interface News {
    news: NewsItem[] | null;
    events: EventItem[] | null;
}

export interface NewsItem {
    id: string;
    title: string;
    body: string;
    date: Date;
    link: string;
}

export interface EventItem {
    id: string;
    title: string;
    infoLink: string;
    date: Date;
}
