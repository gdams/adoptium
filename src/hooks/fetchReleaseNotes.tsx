import { VersionMetaData } from '.';
import { useEffect, useState } from 'react';
import axios from 'axios';

const baseUrl = 'https://api.adoptium.net/v3/info/release_notes';

export function fetchReleaseNotesForVersion(
    isVisible: boolean,
    version: any,
    sortReleaseNotesByCallback?: Function,
): ReleaseNoteDataBag | null {
    if (!version) {
        return null
    }

    const [releaseNotes, setReleaseNotes] = useState<ReleaseNoteDataBag | null>(null);

    useEffect(() => {
        if (isVisible) {
        (async () => {
            const url = `${baseUrl}/${version}`;

            await axios.get(url.toString())
                .then(function (response) {
                    let result = response.data;
                    if(sortReleaseNotesByCallback) sortReleaseNotesByCallback(result);
                    let releaseNoteDataBag: ReleaseNoteDataBag = {
                        releaseNoteAPIResponse: result, 
                        isValid: (result.release_notes !== null)
                    };
                    setReleaseNotes(releaseNoteDataBag);
                })
                .catch(function (error) {
                    let releaseNoteDataBag: ReleaseNoteDataBag = {
                        isValid: false
                    };
                    setReleaseNotes(releaseNoteDataBag);
                });
        })();
        }
    }, [isVisible, version]);

    return releaseNotes;
}

export interface ReleaseNoteDataBag {
    releaseNoteAPIResponse?: ReleaseNoteAPIResponse
    isValid: boolean
}

export interface ReleaseNoteAPIResponse {
    id: string;
    release_name: string;
    release_notes: ReleaseNote[];
    vendor: string;
    version_data: VersionMetaData;
}

export interface ReleaseNote {
    id: string;
    link: URL;
    title: string;
    backportOf?: string;
    priority?: string;
    component?: string;
    subcomponent?: string;
    type?: string;
}
