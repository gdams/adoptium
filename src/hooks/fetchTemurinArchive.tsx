import moment from "moment";
import { fetchExtension } from '../util/fetchExtension';

const baseUrl = 'https://api.adoptium.net/v3';

let releases: TemurinReleases[] = []

export async function getAssetsForVersion(
  version: number,
  releaseType: any,
  numBuilds: number,
  buildDate: Date
): Promise<TemurinReleases[] | null> {
  let url = new URL(`${baseUrl}/assets/feature_releases/${version}/${releaseType}?vendor=eclipse`);
  if (numBuilds) {
    url.searchParams.append('page_size', numBuilds.toString());
  }
  if (buildDate) {
    url.searchParams.append('before', moment(buildDate).format('Y-MM-DD'));
  }
  releases = []
  const response = await getPkgs(url)
  const packages = JSON.parse(response)
  let pkgsFound: TemurinReleases[] = []
  for (let pkg of packages) {
      pkgsFound.push(pkg);
  }
  return renderReleases(pkgsFound);
}

async function getPkgs(url: URL) {
  let response = await fetch(url)
  return response.text();
}

function renderReleases(pkgs) {
  pkgs.forEach((aRelease) => {
    const release: TemurinReleases = {
        release_name: aRelease.release_name,
        release_link: aRelease.release_link,
        timestamp: aRelease.timestamp,
        platforms: {},
    };

    aRelease.binaries.forEach((
      aReleaseAsset: APIResponse) => {
        const platform = `${aReleaseAsset.os}-${aReleaseAsset.architecture}`
  
        // Skip this asset if its platform could not be matched (see the website's 'config.json')
        if (!platform) {
          return;
        }
  
        // Skip this asset if it's not a binary type we're interested in displaying
        const binary_type = aReleaseAsset.image_type.toUpperCase();
        if (binary_type == 'SOURCES') {
          release.source_url = new URL(aReleaseAsset.package.link);
        }
        if (!['INSTALLER', 'JDK', 'JRE'].includes(binary_type)) {
          return;
        }
  
        if (!release.platforms[platform]) {
          release.platforms[platform] = {
            assets: [],
          };
        }
  
        let binary_constructor: ReleaseAsset = {
          os: aReleaseAsset.os,
          architecture: aReleaseAsset.architecture,
          type: binary_type,
          link: aReleaseAsset.package.link,
          checksum: aReleaseAsset.package.checksum,
          size: Math.floor(aReleaseAsset.package.size / 1000 / 1000),
          extension: fetchExtension(aReleaseAsset.package.link.toString())
        };
  
        if (aReleaseAsset.installer) {
          binary_constructor.installer_link = aReleaseAsset.installer.link;
          binary_constructor.installer_checksum = aReleaseAsset.installer.checksum;
          binary_constructor.installer_size =  Math.floor(aReleaseAsset.installer.size / 1000 / 1000);
          binary_constructor.installer_extension = fetchExtension(aReleaseAsset.installer.link.toString())
        }
  
        // Add the new binary to the release asset
        release.platforms[platform].assets.push(binary_constructor);
    });
    releases.push(release);
  })
  return releases
}

export interface TemurinReleases {
  release_name: string;
  release_link: URL;
  source_url?: URL;
  timestamp: Date;
  platforms: {
    [key: string]: {
      assets: ReleaseAsset[];
    };
  };
}

interface ReleaseAsset {
  os: string;
  architecture: string;
  type: string;
  link: URL;
  checksum: string;
  size: number;
  extension: string;
  installer_link?: URL;
  installer_checksum?: string;
  installer_size?: number;
  installer_extension?: string;
}

interface APIResponse {
  os: string;
  architecture: string;
  image_type: string;
  package: {
    link: URL;
    checksum: string;
    size: number;
  };
  installer?: {
    link: URL;
    checksum: string;
    size: number;
  };
}