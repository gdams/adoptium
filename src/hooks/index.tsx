export * from './useOnScreen';
export * from './useAdoptiumContributorsApi';
export * from './fetchMarketplace';
export * from './fetchTemurinReleases';
export * from './fetchTemurinArchive';
export * from './fetchLatestTemurin';
export * from './fetchNews';
export * from './fetchReleaseNotes';

export interface VersionMetaData {
    major: number;
    minor: number;
    security: number;
    pre?: string;
    patch?: number;
    build: number;
    optional?: string;
    adopt_build_number?: number;
    openjdk_version: string;
    semver: string;
};

/**
 * This method compares two versions and returns:
 * ->  0 if versions are equals
 * -> -1 if versionA is lower than versionB
 * ->  1 if versionA is highter than versionB
 * 
 * @param versionA 
 * @param versionB 
 * @returns 
 */
export function compareVersions(versionA: VersionMetaData, versionB: VersionMetaData) : number {
    let comparison = versionA.major - versionB.major;
    if (comparison === 0) {
        comparison = versionA.minor - versionB.minor;
        if (comparison === 0) {
            comparison = versionA.security - versionB.security;
            if (comparison === 0) {
                comparison = versionA.build - versionB.build;
                if (comparison === 0) {
                    comparison = (versionA.adopt_build_number ? versionA.adopt_build_number : 0) - (versionB.adopt_build_number? versionB.adopt_build_number : 0);
                }
            }
        }
    }
    return comparison;
}

export function getVersionAsString(version: VersionMetaData) : string {
    let result = `${version.major}.${version.minor}.${version.security}`;
    if(version.build) {
        result += `+${version.build}`;
        if(version.adopt_build_number) result += `.${version.adopt_build_number}`;
    }
    return result;
}
