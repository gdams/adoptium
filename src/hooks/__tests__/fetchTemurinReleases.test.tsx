import { renderHook } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'
import { loadLatestAssets } from '../fetchTemurinReleases';
import { createMockTemurinReleaseAPI  } from '../../__fixtures__/hooks';
import AxiosInstance from 'axios'
import MockAdapter from 'axios-mock-adapter';

const mock = new MockAdapter(AxiosInstance);
let mockResponse = [createMockTemurinReleaseAPI(false, 'jdk')];

afterEach(() => {
  vi.clearAllMocks();
});

describe('loadLatestAssets', () => {
  it('returns valid JSON', async() => {
    mock.onGet().reply(200, mockResponse);

    renderHook(async() => {
      await loadLatestAssets(8, 'linux', 'x64', 'jdk').then((data) => {
        expect(data).toMatchSnapshot()
      })
    });
  })

  it('source image is processed correctly', async() => {
    mockResponse = [
      createMockTemurinReleaseAPI(false, 'sources'),
      createMockTemurinReleaseAPI(false, 'jdk')
    ];

    mock.onGet().reply(200, mockResponse);

    renderHook(async() => {
      await loadLatestAssets(8, 'linux', 'x64', 'any').then((data) => {
        expect(data).toMatchSnapshot()
      })
    });
  });

  it('returns valid JSON + installer', async() => {
    mockResponse = [
      createMockTemurinReleaseAPI(true, 'jdk'),
      createMockTemurinReleaseAPI(true, 'jre')
    ]

    mock.onGet().reply(200, mockResponse);

    renderHook(async() => {
      await loadLatestAssets(8, 'linux', 'x64', 'jre').then((data) => {
        expect(data).toMatchSnapshot()
      })
    });
  });

  it('verify update the release date if this asset is newer', async() => {
    const r1 = createMockTemurinReleaseAPI(true, 'jdk');
    const r2 = createMockTemurinReleaseAPI(true, 'jdk');
    r2.binary.updated_at.setDate(r2.binary.updated_at.getDate() + 1);

    mockResponse = [r1, r2]

    mock.onGet().reply(200, mockResponse);

    renderHook(async() => {
      await loadLatestAssets(8, 'linux', 'x64', 'jdk').then((data) => {
        expect(data).toMatchSnapshot()
      })
    });
  });

  it('verify that releases are well sorted', async() => {
    const r1 = createMockTemurinReleaseAPI(true, 'jdk');
    r1.binary.architecture = 'x32';
    const r2 = createMockTemurinReleaseAPI(true, 'jdk');
    r2.binary.architecture = 'x64';
    const r3 = createMockTemurinReleaseAPI(true, 'jdk');
    r3.binary.architecture = 'aarch64';
    const r4 = createMockTemurinReleaseAPI(true, 'jdk');
    r4.binary.architecture = 'ppc64le';

    mockResponse = [r1, r2, r3, r4]

    mock.onGet().reply(200, mockResponse);

    renderHook(async() => {
      await loadLatestAssets(8, 'linux', 'x64', 'jdk').then((data) => {
        expect(data).toMatchSnapshot()
      })
    });
  });

  it('pkgsFound to be empty on error', async() => {
    mock.onGet().reply(500);

    renderHook(async() => {
      await loadLatestAssets(8, 'linux', 'x64', 'jdk').then((data) => {
        expect(data).toStrictEqual([])
      })
    });
  })
});
