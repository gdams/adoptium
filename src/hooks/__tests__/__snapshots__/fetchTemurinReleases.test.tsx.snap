// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`loadLatestAssets > returns valid JSON + installer 1`] = `
[
  {
    "architecture": "arch_mock",
    "binaries": [
      {
        "checksum": "checksum_mock",
        "extension": ".tar.gz",
        "installer_checksum": "installer_checksum_mock",
        "installer_extension": ".msi",
        "installer_link": "https://installer_link_mock/",
        "installer_size": 0,
        "link": "https://link_mock/",
        "size": 0,
        "type": "JRE",
      },
    ],
    "os": "os_mock",
    "platform_name": "os_mock-arch_mock",
    "release_date": 2020-01-01T00:00:00.000Z,
    "release_link": "https://release_link_mock/",
    "release_name": "0.0.0",
    "version": {
      "build": 0,
      "major": 0,
      "minor": 0,
      "openjdk_version": "openjdk_version_mock",
      "patch": 0,
      "security": 0,
    },
  },
]
`;

exports[`loadLatestAssets > returns valid JSON 1`] = `
[
  {
    "architecture": "arch_mock",
    "binaries": [
      {
        "checksum": "checksum_mock",
        "extension": ".tar.gz",
        "link": "https://link_mock/",
        "size": 0,
        "type": "JDK",
      },
    ],
    "os": "os_mock",
    "platform_name": "os_mock-arch_mock",
    "release_date": 2020-01-01T00:00:00.000Z,
    "release_link": "https://release_link_mock/",
    "release_name": "0.0.0",
    "version": {
      "build": 0,
      "major": 0,
      "minor": 0,
      "openjdk_version": "openjdk_version_mock",
      "patch": 0,
      "security": 0,
    },
  },
]
`;

exports[`loadLatestAssets > source image is processed correctly 1`] = `
[
  {
    "architecture": "arch_mock",
    "binaries": [
      {
        "checksum": "checksum_mock",
        "extension": ".tar.gz",
        "link": "https://link_mock/",
        "size": 0,
        "type": "JDK",
      },
    ],
    "os": "os_mock",
    "platform_name": "os_mock-arch_mock",
    "release_date": 2020-01-01T00:00:00.000Z,
    "release_link": "https://release_link_mock/",
    "release_name": "0.0.0",
    "version": {
      "build": 0,
      "major": 0,
      "minor": 0,
      "openjdk_version": "openjdk_version_mock",
      "patch": 0,
      "security": 0,
    },
  },
]
`;

exports[`loadLatestAssets > verify that releases are well sorted 1`] = `
[
  {
    "architecture": "aarch64",
    "binaries": [
      {
        "checksum": "checksum_mock",
        "extension": ".tar.gz",
        "installer_checksum": "installer_checksum_mock",
        "installer_extension": ".msi",
        "installer_link": "https://installer_link_mock/",
        "installer_size": 0,
        "link": "https://link_mock/",
        "size": 0,
        "type": "JDK",
      },
    ],
    "os": "os_mock",
    "platform_name": "os_mock-aarch64",
    "release_date": 2020-01-01T00:00:00.000Z,
    "release_link": "https://release_link_mock/",
    "release_name": "0.0.0",
    "version": {
      "build": 0,
      "major": 0,
      "minor": 0,
      "openjdk_version": "openjdk_version_mock",
      "patch": 0,
      "security": 0,
    },
  },
  {
    "architecture": "ppc64le",
    "binaries": [
      {
        "checksum": "checksum_mock",
        "extension": ".tar.gz",
        "installer_checksum": "installer_checksum_mock",
        "installer_extension": ".msi",
        "installer_link": "https://installer_link_mock/",
        "installer_size": 0,
        "link": "https://link_mock/",
        "size": 0,
        "type": "JDK",
      },
    ],
    "os": "os_mock",
    "platform_name": "os_mock-ppc64le",
    "release_date": 2020-01-01T00:00:00.000Z,
    "release_link": "https://release_link_mock/",
    "release_name": "0.0.0",
    "version": {
      "build": 0,
      "major": 0,
      "minor": 0,
      "openjdk_version": "openjdk_version_mock",
      "patch": 0,
      "security": 0,
    },
  },
  {
    "architecture": "x64",
    "binaries": [
      {
        "checksum": "checksum_mock",
        "extension": ".tar.gz",
        "installer_checksum": "installer_checksum_mock",
        "installer_extension": ".msi",
        "installer_link": "https://installer_link_mock/",
        "installer_size": 0,
        "link": "https://link_mock/",
        "size": 0,
        "type": "JDK",
      },
    ],
    "os": "os_mock",
    "platform_name": "os_mock-x64",
    "release_date": 2020-01-01T00:00:00.000Z,
    "release_link": "https://release_link_mock/",
    "release_name": "0.0.0",
    "version": {
      "build": 0,
      "major": 0,
      "minor": 0,
      "openjdk_version": "openjdk_version_mock",
      "patch": 0,
      "security": 0,
    },
  },
  {
    "architecture": "x32",
    "binaries": [
      {
        "checksum": "checksum_mock",
        "extension": ".tar.gz",
        "installer_checksum": "installer_checksum_mock",
        "installer_extension": ".msi",
        "installer_link": "https://installer_link_mock/",
        "installer_size": 0,
        "link": "https://link_mock/",
        "size": 0,
        "type": "JDK",
      },
    ],
    "os": "os_mock",
    "platform_name": "os_mock-x32",
    "release_date": 2020-01-01T00:00:00.000Z,
    "release_link": "https://release_link_mock/",
    "release_name": "0.0.0",
    "version": {
      "build": 0,
      "major": 0,
      "minor": 0,
      "openjdk_version": "openjdk_version_mock",
      "patch": 0,
      "security": 0,
    },
  },
]
`;

exports[`loadLatestAssets > verify update the release date if this asset is newer 1`] = `
[
  {
    "architecture": "arch_mock",
    "binaries": [
      {
        "checksum": "checksum_mock",
        "extension": ".tar.gz",
        "installer_checksum": "installer_checksum_mock",
        "installer_extension": ".msi",
        "installer_link": "https://installer_link_mock/",
        "installer_size": 0,
        "link": "https://link_mock/",
        "size": 0,
        "type": "JDK",
      },
      {
        "checksum": "checksum_mock",
        "extension": ".tar.gz",
        "installer_checksum": "installer_checksum_mock",
        "installer_extension": ".msi",
        "installer_link": "https://installer_link_mock/",
        "installer_size": 0,
        "link": "https://link_mock/",
        "size": 0,
        "type": "JDK",
      },
    ],
    "os": "os_mock",
    "platform_name": "os_mock-arch_mock",
    "release_date": 2020-01-02T00:00:00.000Z,
    "release_link": "https://release_link_mock/",
    "release_name": "0.0.0",
    "version": {
      "build": 0,
      "major": 0,
      "minor": 0,
      "openjdk_version": "openjdk_version_mock",
      "patch": 0,
      "security": 0,
    },
  },
]
`;
